commit 13e6560075607ff354b48556f306bee925328325
Author: Vivek Goyal <vgoyal@redhat.com>
Date:   Wed Nov 1 15:37:22 2017 -0400

    ovl: Put upperdentry if ovl_check_origin() fails
    
    commit 5455f92b54e516995a9ca45bbf790d3629c27a93 upstream.
    
    If ovl_check_origin() fails, we should put upperdentry. We have a reference
    on it by now. So goto out_put_upper instead of out.
    
    Fixes: a9d019573e88 ("ovl: lookup non-dir copy-up-origin by file handle")
    Signed-off-by: Vivek Goyal <vgoyal@redhat.com>
    Signed-off-by: Miklos Szeredi <mszeredi@redhat.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/fs/overlayfs/namei.c b/fs/overlayfs/namei.c
index a12dc10bf726..bc6d5c5a3443 100644
--- a/fs/overlayfs/namei.c
+++ b/fs/overlayfs/namei.c
@@ -630,7 +630,7 @@ struct dentry *ovl_lookup(struct inode *dir, struct dentry *dentry,
 			err = ovl_check_origin(upperdentry, roe->lowerstack,
 					       roe->numlower, &stack, &ctr);
 			if (err)
-				goto out;
+				goto out_put_upper;
 		}
 
 		if (d.redirect) {
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1041_13e6560075607ff354b48556f306bee925328325.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1041_13e6560075607ff354b48556f306bee925328325.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
